core-infrastructure/terraform/dashboards.tf

```diff
@@ -20,22 +21,47 @@ resource "azurerm_portal_dashboard" "mi-dashboard" {
+      popular_school_requests_chart_id    = azurerm_log_analytics_query_pack_query.popular-school-requests-chart.name
       popular_school_requests_chart_query = local.popular-school-requests-chart-query,
       popular_school_requests_chart_title = azurerm_log_analytics_query_pack_query.popular-school-requests-chart.description
 
-      popular_trust_requests_chart_id    = random_uuid.popular-trust-requests-chart-id.result,
+      popular_trust_requests_chart_id    = azurerm_log_analytics_query_pack_query.popular-trust-requests-chart.name
       popular_trust_requests_chart_query = local.popular-trust-requests-chart-query,
       popular_trust_requests_chart_title = azurerm_log_analytics_query_pack_query.popular-trust-requests-chart.description
 
-      popular_local_authority_requests_chart_id    = random_uuid.popular-local-authority-requests-chart-id.result
+      popular_local_authority_requests_chart_id    = azurerm_log_analytics_query_pack_query.popular-local-authority-requests-chart.name
       popular_local_authority_requests_chart_query = local.popular-local-authority-requests-chart-query,
       popular_local_authority_requests_chart_title = azurerm_log_analytics_query_pack_query.popular-local-authority-requests-chart.description
+
+      user_counts_table_id    = azurerm_log_analytics_query_pack_query.user-counts-table.name
+      user_counts_table_query = local.user-counts-table-query,
+      user_counts_table_title = azurerm_log_analytics_query_pack_query.user-counts-table.description
+
+      user_sessions_table_id    = azurerm_log_analytics_query_pack_query.user-sessions-table.name
+      user_sessions_table_query = local.user-sessions-table-query,
+      user_sessions_table_title = azurerm_log_analytics_query_pack_query.user-sessions-table.description
+
+      custom_data_funnel_id    = azurerm_log_analytics_query_pack_query.custom-data-funnel.name
+      custom_data_funnel_query = local.custom-data-funnel-query,
+      custom_data_funnel_title = azurerm_log_analytics_query_pack_query.custom-data-funnel.description
+
+      popular_commercial_resources_id    = azurerm_log_analytics_query_pack_query.popular-commercial-resources.name,
+      popular_commercial_resources_query = local.popular-commercial-resources-query,
+      popular_commercial_resources_title = azurerm_log_analytics_query_pack_query.popular-commercial-resources.description
+
+      feature_requests_chart_id    = azurerm_log_analytics_query_pack_query.feature-requests-chart.name,
+      feature_requests_chart_query = local.feature-requests-chart-query,
+      feature_requests_chart_title = azurerm_log_analytics_query_pack_query.feature-requests-chart.description
   })
 }
 
+locals {
+  failures-query-prefix = replace(replace(file("${path.module}/queries/failures-query-prefix.kql"), "/[\r\n]+/", "\\n"), "\"", "\\\"")
+}
+
 resource "azurerm_portal_dashboard" "oi-dashboard" {
   name                = "${var.environment-prefix}-ebis-dashboard-oi"
```

Comment:
nitpick: could we rename the dashboards {prefix}-oi-dashboard & {prefix}-mi-dashboard. It a pain trying to find them.